apps/search/src/components/ImportProductsToAlgolia/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { Layout } from "@saleor/apps-ui";
+import { Box, Button, Text } from "@saleor/macaw-ui";
+
+import { useAlgoliaConfiguration } from "./useAlgoliaConfiguration";
+import { useBatchUpload } from "./useBatchUpload";
+
+export const ImportProductsToAlgolia = () => {
+  const configuration = useAlgoliaConfiguration();
+  const isConfigured = configuration.type === "configured";
+  const searchProvider = isConfigured ? configuration.provider : null;
+  const { startUpload, uploadState } = useBatchUpload(searchProvider);
+
+  return (
+    <Layout.AppSectionCard
+      footer={
+        isConfigured && (
+          <Box display={"flex"} justifyContent={"flex-end"}>
+            <Button disabled={uploadState.type === "uploading"} onClick={startUpload}>
+              {uploadState.type === "uploading" ? "Importing..." : "Start importing"}
+            </Button>
+          </Box>
+        )
+      }
+      __cursor={uploadState.type === "uploading" ? "wait" : "auto"}
+    >
+      {configuration.type === "loading" ? (
+        <Box>
+          <Text size={5} fontWeight="bold" as={"p"} marginBottom={1.5}>
+            Loading configuration...
+          </Text>
+        </Box>
+      ) : isConfigured ? (
+        <Box>
+          <Text size={5} fontWeight="bold" as={"p"} marginBottom={1.5}>
+            Importing products & variants
+          </Text>
+          <Text as={"p"}>
+            Trigger initial indexing for products catalogue. It can take few minutes.{" "}
+          </Text>
+          <Text marginBottom={5} size={4} fontWeight="bold">
+            Do not close the app - its running client-side
+          </Text>
+        </Box>
+      ) : (
+        <Box>
+          <Text size={5} fontWeight="bold" as={"p"} color={"critical1"} marginBottom={1.5}>
+            App not configured
+          </Text>
+          <Text>Configure Algolia first</Text>
+        </Box>
+      )}
+
+      {uploadState.type === "success" && (
+        <Box marginTop={5}>
+          <Text size={5} fontWeight="bold" as="p">
+            Import finished successfully!
+          </Text>
+        </Box>
+      )}
```

Comment:
Question/suggestion: Is this visible enough? Afaik app-bridge allows apps to show toasts in dashboard.